sktime/forecasting/model_selection/_split.py

```diff
@@ -158,42 +175,100 @@ def _get_end(y, fh):
     return end
 
 
-def _check_window_lengths(y, fh, window_length, initial_window):
+def _check_window_lengths(
+    y: ACCEPTED_Y_TYPES,
+    fh: ForecastingHorizon,
+    window_length: NON_FLOAT_WINDOW_LENGTH_TYPES,
+    initial_window: NON_FLOAT_WINDOW_LENGTH_TYPES,
+) -> None:
     n_timepoints = y.shape[0]
     fh_max = fh[-1]
 
-    if window_length + fh_max > n_timepoints:
-        raise ValueError(
-            f"The `window_length` and the forecasting horizon are incompatible with "
-            f"the length of `y`. Found `window_length`={window_length}, `max(fh)`="
-            f"{fh_max}, but len(y)={n_timepoints}."
-        )
-
-    if initial_window is not None:
-        if initial_window + fh_max > n_timepoints:
+    if is_timedelta_or_date_offset(x=window_length):
+        if y.get_loc(min(y[-1], y[0] + window_length)) + fh_max > n_timepoints:
+            raise ValueError(
+                f"The `window_length` and the forecasting horizon are incompatible "
```

Comment:
Here, I think it would be good to spell out the precise condition in the error message (which is violated)

Reply:
Added these two lines:
```
f"It is required that the window length plus maximum forecast horizon "
f"is smaller than the length of the time series `y` itself."
```
I have also extracted common error text into variables which are used twice.

Reply:
yes, ok. May be improved by ensuring this is clearly linked to functino/method inputs, but good enough.